src/services/auth/error.ts

```diff
@@ -0,0 +1,33 @@
+import { StatusCodes } from 'http-status-codes';
+
+import { ErrorFactory } from '@graasp/sdk';
+
+const PLUGIN_NAME = 'graasp-plugin-auth';
+
+export const GraaspAuthError = ErrorFactory(PLUGIN_NAME);
+
+export class InsufficientPermission extends GraaspAuthError {
+  constructor(data?: unknown) {
+    super(
+      {
+        code: 'GAUTHERR001',
+        statusCode: StatusCodes.FORBIDDEN,
+        message: 'User has insufficient permission to perform this action',
+      },
+      data,
+    );
+  }
+}
+
+export class InvalidatedMember extends GraaspAuthError {
```

Comment:
```suggestion
export class NotValidatedMember extends GraaspAuthError {
```
Pour moi même en français il y a une nuance entre "invalidé" il a été mécaniquement été invalidé (eg banni) alors que "pas validé" c'est qu'il n'a pas encore été validé.